lib/spack/docs/build_settings.rst

```diff
@@ -209,6 +209,170 @@ Specific limitations include:
   then Spack will not add a new external entry (``spack config blame packages``
   can help locate all external entries).
 
+
+.. _manually-adding-external-packages:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Manually Adding External Packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section discusses how to add external packages to the ``packages.yaml``
+file when modules depend on other modules or when Spack is unable to compute
+the prefix. When using modules, Spack will attempt to parse the ``module show``
+output in order to determine all relevant settings and variables. Reasons why
+this may not work automatically are
+
+* the wrong package name,
+* missing module dependencies,
+* nonstandard paths, and
+* the use of metamodules, i.e., modules whose only purpose is to load other
+  modules.
+
+For certain libraries there exist multiple implementations (e.g., MPI). If you
+modify the ``packages.yaml`` file, then you must make sure to select the
+correct package name. For example, the package ``mpich`` can be used only for
+the vanilla MPICH implementation but not for Cray MPICH; Cray MPICH has its own
+package called ``cray-mpich``. If the wrong module name is picked, this can
+cause errors later, e.g., Spack may compute a wrong prefix.
+
+The case of missing dependencies will be discussed based on the real-world
```

Comment:
I think it would be useful to add minor section headers for each of the cases you describe initially (e.g. have a section titled "Missing dependencies").